processor/samplingprocessor/processor.go

```diff
@@ -0,0 +1,64 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package samplingprocessor
+
+import (
+	"context"
+	"math/rand"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+type throughputMeasurementProcessor struct {
+	logger          *zap.Logger
+	dropCutOffRatio float64
+}
+
+func newThroughputMeasurementProcessor(logger *zap.Logger, cfg *Config) *throughputMeasurementProcessor {
+	return &throughputMeasurementProcessor{
+		logger:          logger,
+		dropCutOffRatio: cfg.DropRatio,
+	}
+}
+
+func (tmp *throughputMeasurementProcessor) processTraces(_ context.Context, td ptrace.Traces) (ptrace.Traces, error) {
+	//#nosec G404 -- randomly generated number is not used for security purposes. It's ok if it's weak
+	if rand.Float64() <= tmp.dropCutOffRatio {
+		return ptrace.NewTraces(), nil
+	}
+
+	return td, nil
+}
+
+func (tmp *throughputMeasurementProcessor) processLogs(_ context.Context, ld plog.Logs) (plog.Logs, error) {
+	//#nosec G404 -- randomly generated number is not used for security purposes. It's ok if it's weak
+	if rand.Float64() <= tmp.dropCutOffRatio {
```

Comment:
May be a non-concern, but just confirming its all good that this is an approximation right? Just thinking how batching works, which is something I'm not entirely familiar with when it comes from receivers => processors. 

i.e. a drop ratio of .6 
1st emission: 100 logs - rand.Float64 == .5 (drop)
2nd emission: 50 logs - rand.Float64 == .7 (keep)
3rd emission: 100 logs - rand.Float64 == .45 (drop) 

would theoretically give a sample of logs that is only 20% of the true value right(50/250)? Statistics would probably eventually get us to the correct ratio given enough time but unless these go through a batch processor it may be slightly more inaccurate on first couple samples, which might be perfectly fine.

I believe this is probably fine as long as users know that this value is an estimated approximation. Just trying to wrap my head around an example and thinking through some cases.